crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -376,16 +383,16 @@ impl<'a, T: GitChangeDetector> FilterResolver<'a, T> {
                 {
                     // if we are including dependents or dependencies, and we are not excluding
                     // ourselves, then we should add ourselves to the list of packages
-                    walked_dependencies.insert(package);
+                    walked_dependencies.insert(package, reason);
                 } else if !selector.include_dependencies && !selector.include_dependents {
                     // if we are neither including dependents or dependencies, then
                     // add  to the list of cherry picked packages
-                    cherry_picked_packages.insert(package);
+                    cherry_picked_packages.insert(package, reason);
                 }
             }
         }
 
-        let mut all_packages = HashSet::new();
+        let mut all_packages = HashMap::new();
```

Comment:
Especially here I feel like it would be useful to indicate if there are multiple reasons a package was in scope and currently the reason is dependent on the ordering of the `extend` calls.